application/models/watch_test.go

```diff
@@ -161,3 +161,39 @@ func (ms *ModelSuite) TestWatch_GetSetLocation() {
 	ms.InDelta(newLoc.Latitude.Float64, got.Latitude.Float64, 0.0001, "latitude doesn't match")
 	ms.InDelta(newLoc.Longitude.Float64, got.Longitude.Float64, 0.0001, "longitude doesn't match")
 }
+
+func (ms *ModelSuite) TestWatch_matchesPost() {
+	posts := createPostFixtures(ms.DB, 1, 0, false)
+	watches := createWatchFixtures(ms.DB, createUserFixtures(ms.DB, 1).Users)
+
+	dest, err := posts[0].GetDestination()
+	ms.NoError(err)
+	ms.NoError(watches[0].SetLocation(*dest))
```

Comment:
Watches will have different Location instances than posts, right? So, would it be better to create a separate Location just for the Watch that has similar values as the Post's Location?